### ISSUE TITLE
DrawTexOES:2224 error 0x501

### ISSUE THREAD

**Issue by Alexander-Hjelm** [shashlik/shashlik-build#50]
Running on Ubuntu 17.10 with the following commands:

 /opt/shashlik/bin/shashlik-install ~/Downloads/app-debug.apk
sudo /opt/shashlik/bin/shashlik-run espremea.app-debug espremea.app-debug.apk

Results in a black screen and the following output (I'm happy to provide logs, just tell me where they are / what command to run):
```
starting emulator
emulator: WARNING: system partition size adjusted to match image file (750 MB > 200 MB)

emulator: WARNING: data partition size adjusted to match image file (750 MB > 200 MB)

setting icon3 
Creating filesystem with parameters:
    Size: 69206016
    Block size: 4096
    Blocks per group: 32768
    Inodes per group: 4224
    Inode size: 256
    Journal blocks: 1024
    Label: 
    Blocks: 16896
    Block groups: 1
    Reserved block group size: 7
Created filesystem with 11/4224 inodes and 1302/16896 blocks
../../sdk/emulator/opengl//host/libs/Translator/GLES_CM//GLEScmImp.cpp:glDrawTexOES:2224 error 0x501
../../sdk/emulator/opengl//host/libs/Translator/GLES_CM//GLEScmImp.cpp:glDrawTexOES:2224 error 0x501
../../sdk/emulator/opengl//host/libs/Translator/GLES_CM//GLEScmImp.cpp:glDrawTexOES:2224 error 0x501
```

**Comment by daffamumtaz2361** [comment]
hi, i have the exact same problem. could anyone help? does there is a fix or workaround?

**Comment by warlisonk360** [comment]
First Download in the Terminal:
sudo apt-get install kde-baseapps-bin
Open Terminal direct in folder Download is paste:
/opt/shashlik/bin/shashlik-install nameapp.apk

Worked For me

**Comment by Alexander-Hjelm** [comment]
@warlisonk360 
Hum, kde-baseapps-bin is already latest version for me. I installed kde-baseapps as well for curiosity but it made no difference.

But kde-baseapps is a whole set of applications for KDE if I'm not mistaken? Any idea which of these apps specifically that shashlik requires? Or does shashlike require that I run the KDE desktop environment? I'm currently on XFCE myself.

**Comment by daffamumtaz2361** [comment]
@warlisonk360
I won't install tons of kde libraries. And Alexander-Hjelm says it doesn't work anyway. What i know is shashlik use kde for pop up when an app is installed.

I'm currently on linux mint 18.3 Cinnamon.

**Comment by warlisonk360** [comment]
@Alexander-Hjelm  @Zzombiee2361 
ok, shashlik require a KDE Plasma 5 environment.

**Comment by daffamumtaz2361** [comment]
@warlisonk360
I don't think the problem is from the desktop environment. It seems that the problem came from OpenGL renderer.

**Comment by warlisonk360** [comment]
@Zzombiee2361 
I use Mint 18.3 cinnamon and no have problems with shashlik

**Comment by daffamumtaz2361** [comment]
@warlisonk360 
Exactly like i said, it's not the desktop environment. (Afaik) shashlik only use kde popup to inform user that an app installation is done, just that. The bug came from OpenGL that failed to render the interface.

**Comment by tulakalbeyo** [comment]
i'm using xubuntu 18.04 LTS (bionic beaver).

i installed `kde-baseapps-bin`,

> `After this operation, 273 MB of additional disk space will be used`

ok 🤷 screw it

after everything installs succesfully, i do

`/opt/shashlik/bin/shashlik-run org.tasks org.tasks.apk`

still, no dice, i continue to get the `GLEScmImp.cpp` error

> ` ../../sdk/emulator/opengl//host/libs/Translator/GLES_CM//GLEScmImp.cpp:glDrawTexOES:2224 error 0x501`

now what? what's next?

**Comment by mithodin** [comment]
Same problem here. Installing the packages did nothing.

**Comment by bushrat011899** [comment]
Found a workaround:
Edit 

- /opt/shashlik/android/config.ini
- change `hw.gpu.enabled` to `no`
- change `hw.cpu.arch` to `x86_64`

**Comment by samryzal** [comment]
i still error, any one get solved ?

**Comment by Alexander-Hjelm** [comment]
@samryzal nah, I gave up long ago. Seems this project has been abandoned since a few years back :(
